### PR TITLE
Bug 1505402 - RFC128 updates

### PR DESCRIPTION
**Pull request by djmitche**

### CODE REVIEW

**Comment by djmitche** [comment]
Java's failing, but it's my understanding that's an unrelated issue (it's failing on master too)

**Comment by owlishDeveloper** [comment]
go build is failing too

File: src/test/java/org/mozilla/taskcluster/urls/URLsTest.java [hunk 1]
```diff
@@ -60,10 +60,18 @@ public static String testFunc(String function, URLProvider urlProvider, String[]
...
         case "schema":
             return urlProvider.schema(args[0], args[1]);
+        case "apiReferenceSchema":
+            return urlProvider.apiReferenceSchema(args[0]);
+        case "exchangesReferenceSchema":
+            return urlProvider.exchangesReferenceSchema(args[0]);
+        case "apiManifestSchema":
+            return urlProvider.apiManifestSchema(args[0]);
+        case "metadataMetaschema":
+            return urlProvider.metadataMetaschema();
```
Comment (owlishDeveloper) [inline src/test/java/org/mozilla/taskcluster/urls/URLsTest.java]: your java build is failing because of this line

Comment (owlishDeveloper) [reply src/test/java/org/mozilla/taskcluster/urls/URLsTest.java]: needs a string passed in

Comment (djmitche) [reply src/test/java/org/mozilla/taskcluster/urls/URLsTest.java]: Thanks, silly me.  Fix pushed.

**Comment by owlishDeveloper** [comment]
java build is all good now

go build, however:
`WARNING: staticcheck, gosimple and unused are all set, using megacheck instead`
whatever that means (sorry for jumping in like this lol)

**Comment by djmitche** [comment]
Ah,
```
tcurls.go:64:6:warning: func ApiReferenceSchema should be APIReferenceSchema (golint)
tcurls.go:74:6:warning: func ApiManifestSchema should be APIManifestSchema (golint)
```

**Comment by owlishDeveloper** [comment]
Just to let everybody know - I pulled the branch and ran the java tests locally, they are all nice and green

**Comment by djmitche** [comment]
v12.0.0

**Pull request merged**